pkg/services/libraryelements/api.go

```diff
@@ -56,7 +62,28 @@ func (l *LibraryElementService) createHandler(c *contextmodel.ReqContext) respon
 			cmd.FolderID = folder.ID
 		}
 	}
-	element, err := l.CreateElement(c.Req.Context(), c.SignedInUser, cmd)
+
+	allowed, err := l.AccessControl.Evaluate(
+		c.Req.Context(),
+		c.SignedInUser,
+		ac.EvalAny(
+			ac.EvalPermission(
+				ActionLibraryPanelsCreate,
+			),
+			ac.EvalPermission(
+				dashboards.ActionFoldersWrite,
+				dashboards.ScopeFoldersRoot+":id:"+fmt.Sprint(cmd.FolderID),
+			),
+		),
```

Comment:
The access control middleware for this endpoint already checks `ActionLibraryPanelsCreate`, so this check will always pass, as user needs `ActionLibraryPanelsCreate` to get to this point. What is the intention? Do we want to make sure that they also have access to the folder? Or would it maybe make sense to scope the `ActionLibraryPanelsCreate` action to folders?